dart/lib/src/sentry_client.dart

```diff
@@ -103,8 +106,7 @@ class SentryClient {
         return _sentryId;
       }
     }
-
-    return _options.transport.send(preparedEvent);
+    return await _options.transport.sendSentryEvent(preparedEvent);
```

Comment:
What if we created an envelope with a single event and piped it through to `captureEnvelope`?

Reply:
We had this in the beginning, but this broke many tests that rely on a mock transport handling events and we would also need to implement deserialization of Events form Envelopes.

Reply:
iOS does the same, encapsulation of an event or whatsoever is within the transport for testing, if we'd like to have a single method that takes an envelope to be sent, for testing, we'd need to implement the envelope deserialization, iOS suffers from the same issue I'd say.
So as a quick win, we've decided to do pretty much the same, so we can unit test easily

Reply:
But it's a rather poor public API to have these two methods. 

If we're basing on other SDKs, lets take a look at some that do this right. We should be addressing iOS then also to have a single method: captureEnvelope.

If we're doing this so this PR doesn't grow indefintely and before releasing this we have another PR fixing the tests, I think that'd be best.

Adding a method to the interface is anyway a breaking change given implementors now need to implement it so anyway this will require a major bump

Reply:
Sounds good to me.